### ISSUE TITLE
Even after manual installation, no thumbnails appear in Windows Explorer

### ISSUE THREAD

**Issue by ThomasLandsberger** [winseros/WebpPreviewForWindowsExplorer#1]
I downloaded the installer and ran it, and appeared to succeed, but still no webp thumbnails in Windows Explorer.  Then I tracked down the install location and did a manual installation according to the ReadMe, but no change.  See attached screenshots.  Running on a developer machine with Windows 10 Pro and Visual Studio installed.  Any idea what could be going wrong?

![System](https://user-images.githubusercontent.com/86232546/122746496-f8b29780-d281-11eb-96e0-d932d54d99d1.png)

![ManualInstallation](https://user-images.githubusercontent.com/86232546/122746102-95c10080-d281-11eb-8891-5a88945fec79.png)
![NoWebpThumnails](https://user-images.githubusercontent.com/86232546/122746114-99ed1e00-d281-11eb-9ea1-84e7e7c4834b.png)

**Comment by winseros** [comment]
Hi, @ThomasLandsberger. Could you share one of the images that don't show up?

**Comment by SingingDalong** [comment]
I'm experiencing the same problem. Windows 10 LTSC 2021
https://we.tl/t-k0wtzLC3NT

**Comment by winseros** [comment]
The link in your post leads to an unknown junk site so I'll consider your message fishing.

**Comment by SingingDalong** [comment]
> The link in your post leads to an unknown junk site so I'll consider your message fishing.

The link is just a webp file.

**Comment by winseros** [comment]
Ok then. Just want to clarify: do you have any applications, that associate the `.webp` extension with themselves?

**Comment by SingingDalong** [comment]
> Ok then. Just want to clarify: do you have any applications, that associate the `.webp` extension with themselves?

It's Chrome.

**Comment by winseros** [comment]
Give me some days - I'll make the installer more aggressive so it would override any existing `.webp` associations.

**Comment by SingingDalong** [comment]
> Give me some days - I'll make the installer more aggressive so it would override any existing `.webp` associations.

Thank you for your effort.

**Comment by winseros** [comment]
Hi. Having tested in different environments, I figured out the latest `Windows 10` builds have native `.webp` support. So there is no more need to use any extensions. I tested on a number of builds starting from `20H1`.

![image](https://user-images.githubusercontent.com/6763540/143819134-033cfc55-c6f6-425e-a052-e13d79f3c97f.png)

**Comment by SingingDalong** [comment]
> Hi. Having tested in different environments, I figured out the latest `Windows 10` builds have native `.webp` support. So there is no more need to use any extensions. I tested on a number of builds starting from `20H1`.
> 
> ![image](https://user-images.githubusercontent.com/6763540/143818983-d345849b-406d-4410-bbe4-65688c59ac56.png)

I use 21H2 (LTSC 2021). However, Edge deleted it. Do I need to install Edge?

**Comment by winseros** [comment]
Nope. Edge is not responsible for preview. You can check the system registry - the thumbnail handlers are configured there.

There should be a key and value 
```
[HKEY_CLASSES_ROOT\.webp\ShellEx\{e357fccd-a995-4576-b01f-234630154e96}]
@="{C7657C4A-9F68-40fa-A4DF-96BC08EB3551}"
```

**Comment by winseros** [comment]
Here is a registry script that should activate native support of `.webp` if the machine is capable of it.

```
Windows Registry Editor Version 5.00

[-HKEY_CURRENT_USER\SOFTWARE\Classes\.webp]

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\.webp]

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\.webp\ShellEx]

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\.webp\ShellEx\{BB2E617C-0920-11d1-9A0B-00C04FC2D6C1}]
@="{3F30C968-480A-4C6C-862D-EFC0897BB84B}"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\.webp\ShellEx\{e357fccd-a995-4576-b01f-234630154e96}]
@="{C7657C4A-9F68-40fa-A4DF-96BC08EB3551}"
```

You put it into a text file with the `.reg` extension (i.e. `file.reg`) and execute the file. System restart might be required for the changes to get applied.

**Comment by winseros** [comment]
Here is a registry script that should activate native support of `.webp` if the machine is capable of it.

```
Windows Registry Editor Version 5.00

[-HKEY_CURRENT_USER\SOFTWARE\Classes\.webp]

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\.webp]

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\.webp\ShellEx]

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\.webp\ShellEx\{BB2E617C-0920-11d1-9A0B-00C04FC2D6C1}]
@="{3F30C968-480A-4C6C-862D-EFC0897BB84B}"

[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\.webp\ShellEx\{e357fccd-a995-4576-b01f-234630154e96}]
@="{C7657C4A-9F68-40fa-A4DF-96BC08EB3551}"
```

You put it into a text file with the `.reg` extension (i.e. `file.reg`) and execute the file. System restart might be required for the changes to get applied.

**Comment by ThomasLandsberger** [comment]
I ran the script, got confirmation that the keys were added to the registry, and then restarted my machine.  Unfortunately, .webp files are still not previewed in Windows File Explorer, just showing the IrfanView cat as before.  My system is running Windows 10 Pro 21H1, OS Build 19043.1348.

**Comment by winseros** [comment]
`IrfanView` - that's another story. That guy must be pulling the blanket over and applying its own configs. If you could provide how the registry for the `.webp` extension looks, I'd be able to give more details.

![image](https://user-images.githubusercontent.com/6763540/143826227-b3895569-f24f-4edb-9d39-6ef90004816d.png)

**Comment by ThomasLandsberger** [comment]
Following your instructions, this is what I get:  [also updated the system to 21H2, but no change]

Windows Registry Editor Version 5.00

[HKEY_CLASSES_ROOT\.webp]
"ContentType"="image/webp"
@="webpfile"
"Content Type"="image/webp"
"PerceivedType"="image"

[HKEY_CLASSES_ROOT\.webp\OpenWithList]

[HKEY_CLASSES_ROOT\.webp\OpenWithList\PhotoViewer.dll]
@=""

[HKEY_CLASSES_ROOT\.webp\OpenWithProgids]
"ChromeHTML"=""
"MSEdgeHTM"=""
"webpfile"=""

[HKEY_CLASSES_ROOT\.webp\ShellEx]

[HKEY_CLASSES_ROOT\.webp\ShellEx\ContextMenuHandlers]

[HKEY_CLASSES_ROOT\.webp\ShellEx\ContextMenuHandlers\ShellImagePreview]
@="{FFE2A43C-56B9-4bf5-9A79-CC6D4285608A}"

[HKEY_CLASSES_ROOT\.webp\ShellEx\{BB2E617C-0920-11d1-9A0B-00C04FC2D6C1}]
@="{3F30C968-480A-4C6C-862D-EFC0897BB84B}"

[HKEY_CLASSES_ROOT\.webp\ShellEx\{e357fccd-a995-4576-b01f-234630154e96}]
@="{C7657C4A-9F68-40fa-A4DF-96BC08EB3551}"

**Comment by winseros** [comment]
Seems mostly valid. A silly check then. Is this checkbox enabled?

![image](https://user-images.githubusercontent.com/6763540/143837201-9a427a0a-0f1a-4ef4-8c4e-6f5d4705509f.png)

**Comment by ThomasLandsberger** [comment]
Yep, that's checked, and image files like png, gif, jpg are displayed as thumbnails, but not webp

**Comment by winseros** [comment]
I'm running out of ideas. Thumbs cache cleanup?

![image](https://user-images.githubusercontent.com/6763540/143852136-2ef134db-e120-4e98-bd98-5d2a09b13a74.png)

**Comment by ThomasLandsberger** [comment]
No effect.  Sigh.

**Comment by winseros** [comment]
Well.. I found a machine with Windows 10 21H1 with native `.webp` support not working - quite like yours.

What I did to make things work:

1. Ensured no `WebpPreviewForWindowsExplorer` is installed or manually registered. 
2. Run the script to clean the registry:
    ```
    Windows Registry Editor Version 5.00

    [HKEY_LOCAL_MACHINE\SOFTWARE\Classes\.webp]
    @=-

    [HKEY_CURRENT_USER\SOFTWARE\Classes\.webp]
    @=-
    ```
3. Run the [CCCleaner](https://www.ccleaner.com/ccleaner/download/standard) `Thumbnail cache` cleanup.
    
   ![image](https://user-images.githubusercontent.com/6763540/143915339-ffe6897c-59ad-40aa-99eb-83095e6acf97.png)
4. Installed the  `WebpPreviewForWindowsExplorer` using the installer
5. Runned the `CCCleaner` again.

And it worked

**Comment by ThomasLandsberger** [comment]
Your dedication is exemplary, Nikita.  Unfortunately CCleaner fails to install on my system (it quietly fails even though I tried as admin; no trace of it in Windows Security, so it wasn't being blocked, just failed to install).  So I tried your steps with regular Disk Cleanup of thumbnails, but that did not give me webp thumbnails.  Thanks for trying!
